haystack/preview/components/file_converters/html.py

```diff
@@ -58,7 +58,10 @@ def run(self, sources: List[Union[str, Path, ByteStream]]):
                 logger.warning("Failed to extract text from %s. Skipping it. Error: %s", source, conversion_e)
                 continue
 
-            document = Document(text=text, id_hash_keys=self.id_hash_keys)
+            if self.id_hash_keys:
+                document = Document(text=text, id_hash_keys=self.id_hash_keys)
+            else:
+                document = Document(text=text)
```

Comment:
The condition I introduced is not nice.

But if you pass `id_hash_keys=None`,
the correct default value (`["text", "array", "dataframe", "blob"]`) is not used.

I'm open to better solutions...